### PR TITLE
Provide API context earlier to api requests in order to fix missing schedules

### LINKED ISSUE #29546: API request failures caused by flushing queue on logout don't have the same thread safety guarantees as normal failures
Originally spotted in https://github.com/ppy/osu/pull/28991.

If the user logs out with several API requests still in queue (still not yet started), these web requests will not receive an `APIAccess` instance to schedule callbacks back to:

https://github.com/ppy/osu/blob/4ee5a1225e6e6578b717a5a64987130042b1ef19/osu.Game/Online/API/APIRequest.cs#L112

This means that failure callbacks from the flush-cancellation will not be scheduled back to the update thread:

https://github.com/ppy/osu/blob/4ee5a1225e6e6578b717a5a64987130042b1ef19/osu.Game/Online/API/APIRequest.cs#L180-L183

which makes essentially all `Failure` callbacks essentially not safe if they do anything like transform mutation, because the callback will not be automagically scheduled onto the update thread like it normally is.

@ppy/team-client Would appreciate thoughts on how to handle.

Sentry Issue: [OSU-1J8M](https://sentry.ppy.sh/organizations/ppy/issues/55309/?referrer=github_integration)

```
osu.Framework.Graphics.Drawable+InvalidThreadForMutationException: Cannot mutate the Transforms on a Loaded Drawable while not on the update thread. Consider using Schedule to schedule the mutation operation.
  ?, in void Drawable.EnsureMutationAllowed(string action)
  ?, in void Transformable.AddTransform(Transform transform, ulong? customTransformID)
  ?, in TransformSequence<TThis> TransformableExtensions.TransformTo<TThis>(TThis t, Transform transform)
  ?, in TransformSequence<T> TransformableExtensions.FadeTo<T, TEasing>(T drawable, float newAlpha, double duration, in TEasing easing)
  ?, in TransformSequence<T> TransformableExtensions.FadeOut<T, TEasing>(T drawable, double duration, in TEasing easing)
...
(14 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by peppy**
Closes https://github.com/ppy/osu/issues/29546.

### CODE REVIEW

File: osu.Game/Online/API/APIAccess.cs [hunk 1]
```diff
@@ -568,6 +570,8 @@ public void Queue(APIRequest request)
         {
             lock (queue)
             {
+                request.AttachAPI(this);
```
Comment (peppy) [inline osu.Game/Online/API/APIAccess.cs:573]: The only important line for the fix.

**Review by bdach — DISMISSED** [review]
No objections here, if it passes tests. Maybe check on https://github.com/ppy/osu/pull/29656/commits/8ffd4aa82c5e1afd4b4fe56773d6043248b54a12 to make sure you're ok with how I fixed the inspections.

**Comment by peppy** [comment]
Changes look fine 👍 

My rider was mid rebuilding cached so didn't show me these 😓

**Comment by bdach** [comment]
Tests did not pass but https://github.com/ppy/osu/pull/29656/commits/8b04455c29fd41dfab49974f8883acb7ef60d8fc should *probably* help. I didn't do very thorough investigating but a viable explanation is provided in the commit message.

**Pull request merged**